From bb471773f05971f3ab5021a2b05610ccfceab00e Mon Sep 17 00:00:00 2001
From: Tim Orling <TicoTimo@gmail.com>
Date: Tue, 27 Jan 2026 12:33:14 +0000
Subject: [PATCH] * luaL_checkint and luaL_optint were deprecated in lua 5.3 *
 replacement functions are luaL_checkinteger and luaL_optinteger

Upstream-Status: Inappropriate

RPI-Distro repo forks original vlc and applies patches to enable
raspiberry pi support.

Signed-off-by: Tim Orling <TicoTimo@gmail.com>
---
 modules/lua/demux.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/lua/demux.c b/modules/lua/demux.c
index a4ea3af1a..78b91188f 100644
--- a/modules/lua/demux.c
+++ b/modules/lua/demux.c
@@ -52,7 +52,7 @@ struct vlclua_playlist
 static int vlclua_demux_peek( lua_State *L )
 {
     stream_t *s = (stream_t *)vlclua_get_this(L);
-    int n = luaL_checkint( L, 1 );
+    int n = luaL_checkintegereger( L, 1 );
     const uint8_t *p_peek;
 
     ssize_t val = vlc_stream_Peek(s->p_source, &p_peek, n);
@@ -66,7 +66,7 @@ static int vlclua_demux_peek( lua_State *L )
 static int vlclua_demux_read( lua_State *L )
 {
     stream_t *s = (stream_t *)vlclua_get_this(L);
-    int n = luaL_checkint( L, 1 );
+    int n = luaL_checkintegereger( L, 1 );
     char *buf = malloc(n);
 
     if (buf != NULL)
